### ISSUE TITLE
Is Vadam specific to Reinforcement-Learning?

### ISSUE THREAD

**Issue by razvanmarinescu** [emtiyaz/vadam#6]
This should work for any model, right? Not just for RL? I'm a bit confused for why the ActorNetworkEpsilon in the Tensorflow implementation contains code to optimise the policy network. 

I'm trying to use it for StyleGAN-based image reconstruction, i.e. find a distribution over low-dimensional latents z such that G(z) match a given input image. The network is already trained and fixed, it's just the latent that I want to optimise.

Also, why is the design such that the critic network is also exposed to the outside DDPG.py? That is also something RL-specific, right?  

Thanks,
Raz

**Comment by razvanmarinescu** [comment]
I just realised that the RL-specific implementation is only in Tensorflow. 

For Tensorflow v1, do you have a simple, non-RL implementation? e.g. on MNIST

**Comment by emtiyaz** [comment]
No. Actually, as we have written in the paper, the VOGN version works much better than Vadam. A full study is in our followup paper https://github.com/team-approx-bayes/dl-with-bayes, but this is all in PyTorch.

From what I remember, for Tensorflow, it is difficult to implement sum-of-square gradients (which is what VOGN uses, instead of square-of-sum gradients used in Vadam). PyTorch and Jax are much better. Hope this answers your question.

**Comment by razvanmarinescu** [comment]
Thanks for the answer, and sorry for the late reply! I'm moving my model to Pytorch right now, so I should be able to use VOGN after the migration. 

But just to clarify, the VOGN Pytorch version will work for any loss-function of any model, not just RL, right?

**Comment by emtiyaz** [comment]
From what I remember, it currently works for classification and any model. We were going to add other loss function but I don't remember if somebody got around to that (But I could be wrong).
